test/broadcasting.jl

```diff
@@ -0,0 +1,364 @@
+module TestBroadcasting
+
+using Test, DataFrames
+
+refdf = DataFrame(ones(3, 5))
+
+@testset "normal data frame and data frame row in broadcasted assignment - one column" begin
+    df = copy(refdf)
+    df[1] .+= 1
+    @test df.x1 == [2, 2, 2]
+    @test df[2:end] == refdf[2:end]
+
+    dfv = @view df[1:2, 2:end]
+    dfv[1] .+= 1
+    @test dfv.x2 == [2, 2]
+    @test dfv[2:end] == refdf[1:2, 3:end]
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    dfr = df[1, 3:end]
+    dfr[end-1:end] .= 10
+    @test all(Vector(dfr) .== [1, 10, 10])
+    @test df.x3[1] == 1 && df.x4[1] == 10 && df.x5[1] == 10
+
+    df = copy(refdf)
+    df[:, 1] .+= 1
+    @test df.x1 == [2, 2, 2]
+    @test df[2:end] == refdf[2:end]
+
+    dfv = @view df[1:2, 2:end]
+    dfv[:, 1] .+= 1
+    @test dfv.x2 == [2, 2]
+    @test dfv[2:end] == refdf[1:2, 3:end]
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    df = copy(refdf)
+    df[1] .+= [1, 1, 1]
+    @test df.x1 == [2, 2, 2]
+    @test df[2:end] == refdf[2:end]
+
+    dfv = @view df[1:2, 2:end]
+    dfv[1] .+= [1, 1]
+    @test dfv.x2 == [2, 2]
+    @test dfv[2:end] == refdf[1:2, 3:end]
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    dfr = df[1, 3:end]
+    dfr[end-1:end] .= [10, 10]
+    @test all(Vector(dfr) .== [1, 10, 10])
+    @test df.x3[1] == 1 && df.x4[1] == 10 && df.x5[1] == 10
+
+    df = copy(refdf)
+    df[:, 1] .+= [1, 1, 1]
+    @test df.x1 == [2, 2, 2]
+    @test df[2:end] == refdf[2:end]
+
+    dfv = @view df[1:2, 2:end]
+    dfv[:, 1] .+= [1, 1]
+    @test dfv.x2 == [2, 2]
+    @test dfv[2:end] == refdf[1:2, 3:end]
+    @test all(Matrix(df[1:2, 1:2]) .== 2)
+
+    df = copy(refdf)
+    dfv = @view df[1:2, 2:end]
+    dfr = df[1, 3:end]
+    @test_throws DimensionMismatch df[1] .= rand(3, 1)
```

Comment:
This block tests single-column matrices, but the code above only tests vectors. Maybe change this block to tests vectors, but put the whole set in a loop, wrapping vectors in a function which is `identity` for the first run, and `reshape(x, :, 1)` for the second run?

(Same below.)

Reply:
I am not fully clear what this would achieve, all operations on `reshape(x, :, 1)` if `x` is a vector should throw an error. I added the test cases with `reshape` (that throw an error) and with some broadcasted vector transformation (that work OK).

Reply:
Woops, yes, that's not the right place for this comment. Actually I should have said that in the test set below, column vectors and single-column matrices should be tested when assigning to a data frame. Does that make sense?

Reply:
Yes :) - I already added column vectors. I will add single column matrices for safety (but I test general matrices and this should be handled correctly by Base).

Reply:
OK I see you've added a test for that.